### ISSUE TITLE
prefer-const add option onlyPrimitive

### ISSUE THREAD

**Issue by Moeriki** [eslint/eslint#3593]
Dear all,

I'd like to propose an option for the *prefer-const* rule.

I really like enforcing const on unchanged primitives. Not so much on unchanged objects. Because const doesn't protect object members I find this an illogical thing to do.

Proposed option: *onlyPrimitive*. This would allow the following code

```json
{
    "rules": {
        "prefer-const": [1, { "onlyPrimitive": true }]
    }
}
```

```javascript
let x = {};
```

I'd like to hear your thoughts on this.
I already made an attempt to implement this change: #3578

CC @mysticatea

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
Sounds like a reasonable request but I dont use ES6 a lot so would rely on @eslint/eslint-team for more feedback here.

**Comment by nzakas** [comment]
I'm not quite agreeing with the rationale for such a change. `const` is about freezing identifier bindings, not about freezing values. In that vein, this suggestion draws an arbitrary distinction that I'm worried would be confusing for others.

Also, we can't staticly determine primitives unless you're just talking about literals, which would make it more arbitrary.

Unless I'm missing something (feel free to let me know), I'm :-1: on this.

**Comment by mysticatea** [comment]
I prefer this rule reports non primitive as well.
Most variables in js codes of my job are `const`.

But I comprehend this request. The `const` variables are used to be frozen completely in some language (C, C++, C#, ...). So  to use `const` for mutable objects can confuse developers who are familiar with those languages.

**Comment by nzakas** [comment]
I'm inclined to suggest you create a custom rule for this use case.

**Comment by Moeriki** [comment]
Fair enough. Thanks for your feedback all.
